packages/dashboard/src/components/widgets/primitives/input/hooks/useWritableSiteWiseProperty.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  DescribeAssetCommand,
+  IoTSiteWiseClient,
+  ListTagsForResourceCommand,
+  ErrorDetails,
+} from '@aws-sdk/client-iotsitewise';
+import { useContext } from 'react';
+import { ClientContext } from '~/components/dashboard/clientContext';
+
+export const WRITABLE_TAG = 'iot-app-kit:writable';
+
+const useWritableSiteWiseProperty = () => {
+  const client = useContext(ClientContext);
+
+  const describeAsset = (client: IoTSiteWiseClient, assetId: string) =>
+    client.send(new DescribeAssetCommand({ assetId }));
+
+  const getTags = (client: IoTSiteWiseClient, resourceArn: string | undefined) =>
+    client.send(new ListTagsForResourceCommand({ resourceArn }));
+
+  return async ({
+    assetId,
+    propertyId,
+    propertyAlias,
+  }: {
+    assetId: string;
+    propertyId: string;
+    propertyAlias?: string;
+  }): Promise<{ writable: boolean; error: string | undefined }> => {
+    if (!client) return { writable: false, error: 'missing client' };
+    let asset;
+    let assetTags;
+
+    try {
+      asset = await describeAsset(client, assetId);
+    } catch (e) {
+      return { writable: false, error: (e as ErrorDetails).message };
```

Comment:
Non-blocker: Don't cast errors. Use a type guard instead.

Reply:
Good point. We can create a predicate for this.